src/iqm/qiskit_iqm/iqm_provider.py

```diff
@@ -176,13 +182,28 @@ def create_run_request(self, run_input: Union[QuantumCircuit, list[QuantumCircui
         )
         qubit_mapping = {str(idx): qb for idx, qb in self._idx_to_qb.items() if idx in used_indices}
 
-        return self.client.create_run_request(
-            circuits_serialized,
-            qubit_mapping=qubit_mapping,
-            calibration_set_id=calibration_set_id if calibration_set_id else None,
-            shots=shots,
-            options=merged_options['circuit_compilation_options'],
-        )
+        if self._use_default_calibration_set:
```

Comment:
IQMBackend now always corresponds to a specific calibration set, which is always used to run the circuits, and thus `create_run_request` does not allow specifying a calset id. There are the following ways to create a backend: 

- `IQMBackend(client)` -> the backend will always use the calibration set that was default at the time when backend was created

- `IQMBackend(client, calibration_set_id=custom_calset)` -> the backend will always use `custom_calset` 

Using a new server default calset requires creating a new backend, as now explained in documentation. I think no change is needed here.

Reply:
Right, a backend cannot easily automatically track the default calibration on the server, since the transpilation target depends on the calset, and transpilation is a separate operation from run. I guess `transpile` could use a callback to ask the backend to update to the current default calset and Target, but then it could again be out of date when the user calls `run`. I think it's fine like this for now.